src/internal/operators/takeWhile.ts

```diff
@@ -1,8 +1,13 @@
 /** @prettier */
-import { OperatorFunction, MonoTypeOperatorFunction } from '../types';
+import { OperatorFunction, MonoTypeOperatorFunction, Falsy } from '../types';
 import { operate } from '../util/lift';
 import { OperatorSubscriber } from './OperatorSubscriber';
 
+export function takeWhile<T>(predicate: (value: T, index: number) => false, inclusive?: boolean): OperatorFunction<T, never>;
+export function takeWhile<T>(
+  predicate: BooleanConstructor,
+  inclusive?: boolean
+): OperatorFunction<T, Exclude<T, Falsy> extends never ? never : T>;
```

Comment:
issue: Pretty sure that the return type can only be narrowed if `inclusive` is `false`.